### ISSUE TITLE
wgan with gradient penalty

### ISSUE THREAD

**Issue by robotcator** [wiseodd/generative-models#35]
I wrote an wgan with gradient penalty on the cifar dataset. 
https://github.com/robotcator/wgan-gp/tree/wgan-cifar10
it would be useful to integrate wgan-gp to this repo.

The reference:
[pytorch version](https://github.com/caogang/wgan-gp)
[original tensorflow version](https://github.com/igul222/improved_wgan_training)

**Comment by wiseodd** [comment]
Currently there is TF version of WGAN-GP: https://github.com/wiseodd/generative-models/tree/master/GAN/improved_wasserstein_gan.

Feel free to submit pull request if you want to add the Pytorch version.
